### PR TITLE
docs(rh-dialog): update docs

### LINKED ISSUE #927: [docs] Dialog docs need review/cleanup
Please review sections and ensure docs are complete.

Current tabs: Overview, Style, Guidelines, Code. “Overview” heading in the Code tab has no additional description beyond the name of the element and a demo button.

Previous IA Docs update issue: #826

Tag @coreyvickery for final design review.

### PR DESCRIPTION
**Pull request by eyevana**
Closes #927

## What I did

1. Updated the documentation for the dialog element according to the specs outlined in #927.

## Testing Instructions

1. Review latest docs at https://deploy-preview-1001--red-hat-design-system.netlify.app/elements/dialog and verify they match the design specs as well as the docs cleanup guide https://github.com/RedHat-UX/red-hat-design-system/wiki/Docs-Cleanup-Q2-2023-Checklist#code-page.

### CODE REVIEW

File: elements/rh-dialog/docs/dialog-16px-spacer.png
Comment (bennypowers) [inline elements/rh-dialog/docs/dialog-16px-spacer.png:1]: do we need to copy these for each element? can't we just keep one copy for everyone?

**Review by bennypowers — CHANGES_REQUESTED** [review]

File: elements/rh-dialog/docs/dialog-16px-spacer.png
Comment (eyevana) [reply elements/rh-dialog/docs/dialog-16px-spacer.png:1]: Yep, that makes sense. Where's the best spot to store these shared SVGs? `docs/assets/spacing`? Is it cool if  break this out into a post-clean up task and update/verify all elements are referring to the shared spacers?

Comment (bennypowers) [reply elements/rh-dialog/docs/dialog-16px-spacer.png:1]: yeah let's do that

**Review by bennypowers — APPROVED** [review]

File: elements/rh-dialog/docs/dialog-16px-spacer.png
Comment (eyevana) [reply elements/rh-dialog/docs/dialog-16px-spacer.png:1]: https://github.com/RedHat-UX/red-hat-design-system/issues/1005

**Pull request merged**